portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java

```diff
@@ -12,20 +26,65 @@
 /** Servlet responsible for creating new comments. */
 @WebServlet("/new-comment")
 public class NewCommentServlet extends HttpServlet {
+    private long tmpId = 123;
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String name = request.getParameter("name");
     String message = request.getParameter("message");
+    String imageUrl = getUploadedFileUrl(request, "image");
     long timestamp = System.currentTimeMillis();
 
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("name", name);
     commentEntity.setProperty("message", message);
     commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("imageUrl", imageUrl);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
+    
+    Comment comment = new Comment(tmpId, name, message, imageUrl,timestamp);
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comment));
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String InputElementName) {
```

Comment:
If it can return null best to annotate with "@Nullable" (capital N, not sure why comments keeps lower casing it) if possible.

Reply:
Will make the change. Thanks!